runtime/compiler/control/CompilationThread.cpp

```diff
@@ -1594,6 +1595,49 @@ void TR::CompilationInfo::releaseLogMonitor()
       }
    }
 
+bool TR::CompilationInfo::isClassLoaderTransient(void *classLoader)
+   {
+   if (!_transientClassLoaderMonitor)
+      {
+      _transientClassLoaderMonitor = TR::Monitor::create("JIT-TransientClassLoadersMonitors");
```

Comment:
By creating this monitor upfront, you can avoid the if statement which is executed frequently.
Moreover, this code also allows two threads to create 2 monitors at the same time which is not good.